numpy/lib/arraypad.py

```diff
@@ -479,11 +489,8 @@ def _append_mean(arr, pad_amt, num, axis=-1):
             num = None
 
     # Slice a chunk from the edge to calculate stats on
-    end = arr.shape[axis] - 1
     if num is not None:
-        mean_slice = tuple(
-            slice(None) if i != axis else slice(end, end - num, -1)
```

Comment:
Note that this slice is the reverse of what it was before - but `min`, `max`, `mean`, and `med` all are order-invariant anyway